src/cargo/ops/cargo_package.rs

```diff
@@ -248,7 +248,12 @@ pub fn package(ws: &Workspace<'_>, opts: &PackageOpts<'_>) -> CargoResult<Vec<Fi
     }
     let pkgs = ws.members_with_features(specs, &opts.cli_features)?;
 
-    if ws.root().join("Cargo.lock").exists() {
+    let default_lockfile = ws.root().join("Cargo.lock");
```

Comment:
Call `ws.lockfile_path()` and we don't need to duplicate ourselves.
See https://github.com/rust-lang/cargo/pull/14326#discussion_r1702450833